packages/kotti-ui/source/kotti-field-file-upload/types.ts

```diff
@@ -0,0 +1,264 @@
+import { yocoIconSchema } from '@3yourmind/yoco'
+import { Yoco } from '@3yourmind/yoco'
+import { v4 as uuidv4 } from 'uuid'
+import { z } from 'zod'
+
+import { KottiField } from '../kotti-field/types'
+
+export namespace KottiFieldFileUpload {
+	export namespace Status {
+		export enum Common {
+			INVALID = 'INVALID',
+			NOT_STARTED = 'NOT_STARTED',
+			UPLOADED = 'UPLOADED',
+			UPLOADED_WITH_ERROR = 'UPLOADED_WITH_ERROR',
+		}
+		export const commonSchema = z.nativeEnum(Common)
+
+		export enum NonRemote {
+			READY_TO_UPLOAD = 'READY_TO_UPLOAD',
+		}
+		export const nonRemoteSchema = z.nativeEnum(NonRemote)
+
+		export enum Remote {
+			CANCELED = 'CANCELED',
+			ERROR = 'ERROR',
+			UPLOADING = 'UPLOADING',
+		}
+		export const remoteSchema = z.nativeEnum(Remote)
+
+		export const schema = commonSchema.or(nonRemoteSchema).or(remoteSchema)
```

Comment:
according to docs, there is no difference between the or and union

have u experienced anything in type validation that was different when executed